### PR TITLE
8283675: Line not removed from LineChart when series cleared

### PR DESCRIPTION
**Pull request by karthikpandelu**
The issue is present in AreaChart along with the LineChart. Issue is fixed in both the charts as part of this PR.
The line elements in case of Line chart and both line element and fill element in the case of Area charts were not cleared in `makePaths()` method in AreaChart.java. Hence the line and fill area were not getting cleared when series was cleared.

Made changes in code to clear line element and fill element as required in the `makePaths()` method.

Added tests to validate the changes in both LineChart and AreaChart.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issue
 * [JDK-8283675](https://bugs.openjdk.org/browse/JDK-8283675): Line not removed from LineChart when series cleared (**Bug** - P3)

### Reviewers
 * [Andy Goryachev](https://openjdk.org/census#angorya) (@andy-goryachev-oracle - **Reviewer**)
 * [Ajit Ghaisas](https://openjdk.org/census#aghaisas) (@aghaisas - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx.git pull/1214/head:pull/1214` \
`$ git checkout pull/1214`

Update a local copy of the PR: \
`$ git checkout pull/1214` \
`$ git pull https://git.openjdk.org/jfx.git pull/1214/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1214`

View PR using the GUI difftool: \
`$ git pr show -t 1214`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/1214.diff">https://git.openjdk.org/jfx/pull/1214.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jfx/pull/1214#issuecomment-1683488413)

### CODE REVIEW

File: modules/javafx.controls/src/test/java/test/javafx/scene/chart/AreaChartTest.java [hunk 1]
```diff
@@ -497,6 +498,24 @@ public void testSeriesRemoveAnimatedStyleClasses() {
         assertArrayEquals(convertSeriesDataToPoint2D(expectedSeries).toArray(), findDataPointsFromPathLine(ac).toArray());
     }
 
+    //JDK-8283675
+    @Test public void testChartFillRemovedOnClearingSeries() {
```
Comment (andy-goryachev-oracle) [inline modules/javafx.controls/src/test/java/test/javafx/scene/chart/AreaChartTest.java:502]: should we add a similar test for other chart types (even if those do not have the issue)?

**Comment by andy-goryachev-oracle** [comment]
While testing, I noticed another issue (an existing one, since it also appears with the master branch):

when clearing/adding/removing points (use the latest version of the Monkey Tester https://github.com/andy-goryachev-oracle/MonkeyTest), the minor ticks are not always getting updated, resulting in the vertical axis rendered incorrectly:

Screenshot 2023-08-21 at 11 30 37

this issue can be seen with any kind of XYChart (area, bar, bubble, line, scatter, stacked area, stacked bar).

@karthikpandelu would you please check if we already have a similar issue and create one if not?

**Comment by karthikpandelu** [comment]
> @karthikpandelu would you please check if we already have a similar issue and create one if not?

I couldn't find a similar issue, created [JDK-8314754](https://bugs.openjdk.org/browse/JDK-8314754)

File: modules/javafx.controls/src/test/java/test/javafx/scene/chart/AreaChartTest.java (hunk 1 above)
Comment (karthikpandelu) [reply modules/javafx.controls/src/test/java/test/javafx/scene/chart/AreaChartTest.java:502]: Could be useful. Do you think it should be done as part of this PR or should we create separate bug and take it up in test sprint?

Comment (andy-goryachev-oracle) [reply modules/javafx.controls/src/test/java/test/javafx/scene/chart/AreaChartTest.java:502]: sure, we can use the upcoming test sprint to add these tests.  would you please create a ticket?

**Review by andy-goryachev-oracle — APPROVED** [review]
This looks good, with 2 more tickets out of scope for this PR.
Could we have another reviewer to take a look please?

File: modules/javafx.controls/src/test/java/test/javafx/scene/chart/AreaChartTest.java (hunk 1 above)
Comment (karthikpandelu) [reply modules/javafx.controls/src/test/java/test/javafx/scene/chart/AreaChartTest.java:502]: Created [JDK-8314779](https://bugs.openjdk.org/browse/JDK-8283675)

**Review by aghaisas — APPROVED** [review]
The fix looks good to me!

**Pull request closed without merging**